### PR TITLE
adding csxtools recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/csxtools-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:22]:
```suggestion
    - llvm-openmp  # [osx]
    - _openmp_mutex  # [linux]
```

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:16]:
```suggestion
```

**Review by mrakitin — APPROVED** [review]
Thanks, @kbeeperez!

@conda-forge/staged-recipes, I agree to be a maintainer on this feedstock.

Q: Did we do it right that we added the missing `_openmp_mutex` dependency for Linux into the `build` section?

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes anything else before it can be merged?

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:32]:
```suggestion
    - matplotlib-base
```

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:22]: I think we had to add `_openmp_mutex` for other recipes too, so I assume it's the right thing to do here (self-answering my question from the previous round of review.)

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:3]:
```suggestion
```

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:11]:
```suggestion
  sha256: 6c7479be91a2980610f065e656a9186ca5c91e863d96b0d7dceb18c497f55873
```

File: recipes/csxtools/meta.yaml
Comment (mrakitin) [inline recipes/csxtools/meta.yaml:49]:
```suggestion
  summary: Useful Python tools for NSLS-II CSX (23-ID-1) beamline
```

**Review by mrakitin — COMMENTED** [review]
I added a couple of formatting suggestions, otherwise, it's good to go.

File: recipes/csxtools/meta.yaml
Comment (xhochy) [reply recipes/csxtools/meta.yaml:21]: This shouldn't be needed 🤔

Comment (mrakitin) [reply recipes/csxtools/meta.yaml:21]: The easiest way to check it is to remove this dependency and try building it again.

@kbeeperez, can you please comment `_openmp_mutex` out or remove it, so that we could see the failure and use the suggestions from CF folks to fix it?

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes This is ready to merge.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes friends, could this please be merged soon? We plan to test the packages at NSLS-II next week (Wednesday, August 4), it would be great to have the packages from this PR available. Thanks a lot for your cooperation and help!

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes

File: recipes/csxtools/meta.yaml
Comment (xhochy) [inline recipes/csxtools/meta.yaml:36]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]
My previous comment was marked as resolved but not addressed.

**Comment by kbeeperez** [comment]
@xhochy https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=358793&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=957

for the open_mutex dependency

File: recipes/csxtools/meta.yaml
Comment (xhochy) [inline recipes/csxtools/meta.yaml:21]: Can you try
```suggestion
    - libgomp # [linux]
```

**Comment by kbeeperez** [comment]
@conda-forge/staged-recipes this should be ready to go now!

**Review by mrakitin — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by mrakitin** [comment]
Thank you, @xhochy!

**Pull request merged**
